src/hist.jl

```diff
@@ -377,8 +377,8 @@ Fit a histogram to `data`.
   bin. If no weight vector is supplied, each observation has weight 1.
 
 * `edges`: a vector (typically an `AbstractRange` object), or tuple of vectors, that gives
-  the edges of the bins along each dimension. If no edges are provided, these
-  are determined from the data.
+  the edges of the bins along each dimension. If no edges are provided, they are chosen
+  so that approximately `nbins` bins of equal width are constructed along each dimension.
```

Comment:
Would something like this help, do you think?

```suggestion
  so that approximately `nbins` bins of equal width are constructed along each dimension.

!!! note
    In most cases, the number of bins will be `nbins`. However, to ensure that the bins have
    equal width, more or fewer than `nbins` bins may be used.
```

Reply:
Yes, if no one has done it, I would need to check if what you added is exactly true (i.e. in what cases we do a different number of bins). I know that if we have many duplicates we make less bins, but I have not checked if we can make more bins.

@nalimilan + @ararslan => do you know exact rules or I need to check them?

Reply:
I unfortunately don't know the exact rules. Assuming the number of edges is equal to the number of bins, we can produce more in the following contrived example:
```
julia> using StatsBase

julia> fit(Histogram, 1:10; nbins=9)
Histogram{Int64, 1, Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}}
edges:
  1.0:1.0:11.0
weights: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
closed: left
isdensity: false

julia> length(ans.edges[1])
11
```
I don't know though whether the number of edges is the same as the number of bins; I'm not very familiar with the histogram code.

Reply:
Indeed it can be extremely bad:
```
julia> h = fit(Histogram, [fill(1, 10); fill(2, 10)]; nbins=45)
Histogram{Int64, 1, Tuple{StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}}
edges:
  1.0:0.05:2.05
weights: [10, 0, 0, 0, 0, 0, 0, 0, 0, 0  …  0, 0, 0, 0, 0, 0, 0, 0, 0, 10]
closed: left
isdensity: false

julia> length(h.weights)
21
```
In this case your description is OK.